### PR TITLE
Refactor build system to be more CMake-friendly

### PR DESCRIPTION
**Pull request by galpeter**
* All dependencies (except nuttx) are built from IoT.js CMake file.
* IoT.js binary and lib is built directly into the correct folder,
  no need for extra copy

IoT.js-DCO-1.0-Signed-off-by: Peter Gal pgal.u-szeged@partner.samsung.com

### CODE REVIEW

File: cmake/iotjs.cmake [hunk 1]
```diff
@@ -14,147 +14,255 @@
...
+  else()
+    set(${RESULT} ${VALUE_FALSE} PARENT_SCOPE)
+  endif()
+endfunction(find_value)
 
 # System Configuration
-set(IOTJS_PLATFORM_SRC "")
-if(${CMAKE_SYSTEM_NAME} MATCHES "Linux|Tizen")
-    list(APPEND IOTJS_PLATFORM_SRC ${SRC_ROOT}/platform/iotjs_*-linux.c)
+set(IOTJS_PLATFORM_SRC )
```
Comment (chokobole) [inline cmake/iotjs.cmake]: please remove space at the ending before parenthesis.

File: cmake/iotjs.cmake [hunk 2]
```diff
@@ -14,147 +14,255 @@
...
-      set(BOARD_MODULE_SRC
-          "${MODULE_STEM}${PLATFORM_DESCRIPT}-${BOARD_DESCRIPT}.c")
-      if(EXISTS "${BOARD_MODULE_SRC}")
-        list(APPEND IOTJS_MODULE_SRC ${BOARD_MODULE_SRC})
-        set(HAVE_MODULE_IMPL "yes")
-        message(STATUS "Board ${module} module found: ${BOARD_MODULE_SRC}")
-      endif()
+# Module Configuration - listup all possible native C modules
+set(IOTJS_MODULES_ENABLED )
+set(IOTJS_MODULES_DISABLED )
```
Comment (chokobole) [inline cmake/iotjs.cmake]: ditto

File: cmake/iotjs.cmake [hunk 3]
```diff
@@ -14,147 +14,255 @@
...
+  get_filename_component(IOTJS_MODULENAME ${module} NAME_WE)
+  string(SUBSTRING ${IOTJS_MODULENAME} 13 -1 IOTJS_MODULENAME)
+  string(TOUPPER ${IOTJS_MODULENAME} IOTJS_MODULENAME)
+  list(APPEND IOTJS_MODULES_DISABLED ${IOTJS_MODULENAME})
+endforeach()
+
+# Module Configuration - enable only selected modules and add board support
+set(IOTJS_PLATFORM_SUPPORT )
+set(IOTJS_BOARD_SUPPORT )
+set(IOTJS_MODULES_SRC )
```
Comment (chokobole) [inline cmake/iotjs.cmake]: ditto

File: cmake/http-parser.cmake [hunk 4]
```diff
@@ -14,6 +14,27 @@
...
+endif()
+
+set(DEPS_HTTPPARSER deps/http-parser)
+set(DEPS_HTTPPARSER_SRC ${ROOT_DIR}/${DEPS_HTTPPARSER}/)
+ExternalProject_Add(http-parser
+  PREFIX ${DEPS_HTTPPARSER}
+  SOURCE_DIR ${DEPS_HTTPPARSER_SRC}
+  BUILD_IN_SOURCE 0
+  BINARY_DIR ${DEPS_HTTPPARSER}
+  INSTALL_COMMAND ""
```
Comment (chokobole) [inline cmake/http-parser.cmake]: If this `INSTALL_COMMAND`'s default value is empty string, how about just using its defalut value ?

**Comment by glistening** [comment]
@galpeter Basically I agree with you. The build system code was divided and may be duplicated in CMake, python, and Makefile. However, since the build system is related to every developer/contributors involving this repository, it would be better write issue first and broadcast you will make changes into build script. Anyway, I appreciate your effort to enhance build system.

File: cmake/http-parser.cmake (hunk 4 above)
Comment (galpeter) [inline cmake/http-parser.cmake]: If there is no INSTALL_COMMAND specified the external project step always tries to invoke the install part by default. There are multiple ways to disable the install part, this is one of them. However I've reworked this as for the nuttx build there were additional requirements.

**Comment by galpeter** [comment]
@glistening sorry for the big delay I've created an issue and also updated the PR

**Pull request merged**